inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/hook/ConditionalHookAction.java

```diff
@@ -0,0 +1,29 @@
+package rocks.inspectit.oce.core.instrumentation.hook;
+
+import lombok.Value;
+
+@Value
+public class ConditionalHookAction implements IHookAction {
+
+
+    @FunctionalInterface
+    public interface Condition {
+        boolean evaluate(IHookAction.ExecutionContext context);
+    }
+
+    private Condition condition;
+
+    private IHookAction action;
```

Comment:
I dont know how lombok is generating the `@Value` code, but should we add `@NonNull` to the fields?

Reply:
`@Value` generates an all args constructor and therefore forces you to specify the values.